src/qibolab/instruments/qm/sequence.py

```diff
@@ -73,8 +66,12 @@ class QMPulse:
     elements_to_align: Set[str] = field(default_factory=set)
 
     def __post_init__(self):
-        self.element: str = f"{self.pulse.type.name.lower()}{self.pulse.qubit}"
-        self.operation: str = self.pulse.serial
+        pulse_type = self.pulse.type.name.lower()
+        amplitude = format(self.pulse.amplitude, ".6f").rstrip("0").rstrip(".")
+        self.element: str = f"{pulse_type}{self.pulse.qubit}"
+        self.operation: str = (
+            f"{pulse_type}({self.pulse.duration}, {amplitude}, {self.pulse.shape})"
+        )
```

Comment:
> Or is there any other reason?

No other reason, only deduplication. In principle this is not strictly required at the pulse level, because the waveforms are registered separately in the QUA config and you can have multiple pulses using the same waveform, but I think even better if we deduplicate even the pulse section. Also readout pulses are a bit more delicate because they are associated with variables (where we save acquisition) and these need to be deduplicated as well (so that we don't have multiple variables for the "same" readout when unrolling).

Reply:
> No other reason, only deduplication. In principle this is not strictly required at the pulse level, because the waveforms are registered separately in the QUA config and you can have multiple pulses using the same waveform, but I think even better if we deduplicate even the pulse section whenever possible.

I wanted to understand because I still believe that deduplication should happen in the common layer, so it should not be a QM specific feature.
The goal would be to assign an ID per waveform at a higher level.

However, this is something for after 0.2, for the time being it is perfectly to have this feature here.

(I also wanted to understand which pulse attributes you were using to dedup)

Reply:
> Of course the exact string format I used is not relevant. Any kind of id (readable or no) that does not duplicate the `start`/`channel` and is of string type works.

I also don't care about the specific format.
For the sake of compression, you could `hash()` the attributes, but I'm pretty sure it's worth nothing.